src/EFCore.SqlServer/Migrations/SqlServerMigrationsSqlGenerator.cs

```diff
@@ -2391,11 +2381,135 @@ private string Uniquify(string variableName, bool increase = true)
         return _variableCounter == 0 ? variableName : variableName + _variableCounter;
     }
 
+    private IReadOnlyList<MigrationOperation> FixLegacyTemporalAnnotations(IReadOnlyList<MigrationOperation> migrationOperations)
+    {
```

Comment:
```suggestion
    {
        if (migrationOperations.All(o => o[SqlServerAnnotationNames.IsTemporal] as bool? != true))
        {
            return migrationOperations;
        }
```
The majority of the migrations will not involve temporal tables, so we can add a shortcircuit.